### PR TITLE
feat(insured-bridge): Implement speedUpRelay and settleRelay

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

Instant relayer needs to pay relay amount minus realized fee.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol [hunk 1]
```diff
@@ -195,18 +184,41 @@ contract BridgePool is Testable {
...
+        require(relays[depositHash].instantRelayer == address(0), "Relay has already been instant relayed");
         relay.instantRelayer = msg.sender;
-        // TODO: Pull funds from caller.
-        emit RelaySpedUp(_depositHash, msg.sender);
+
+        // Pull relay amount minus fees from caller.
+        IERC20(_depositData.l1Token).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _depositData.amount - _getRealizedFeeAmount(relay.realizedFeePct, _depositData.amount)
```
Comment (chrismaree) [inline packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol:205]: nit: put this on a separate line for readability

File: packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol [hunk 2]
```diff
@@ -333,6 +345,30 @@ contract BridgePool is Testable {
                 .rawValue;
     }
 
+    function _getRealizedFeeAmount(uint64 _realizedFeePct, uint256 _amount) private pure returns (uint256) {
```
Comment (chrismaree) [inline packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol:348]: this computation will likely chance in the next few days but it follows the interface spec for now.

**Review by chrismaree — APPROVED** [review]
looks good! we will need to change the fee logic a bit once we've fully converged on exactly how we want to do that but for now this looks good.

File: packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol (hunk 1 above)
Comment (nicholaspai) [reply packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol:205]: I'm trying to keep the stack small and use fewer intermediate variables. Do you think this gives gas savings either way?

**Pull request merged**